sensibo/unified.js

```diff
@@ -1,235 +1,485 @@
 			}
 
-			platform.log.easyDebug(`${roomName} - Mode: ${mode} - Temperature scales, C: ${'C' in modeCapabilities.temperatures} F: ${'F' in modeCapabilities.temperatures}`)
+			platform.easyDebugInfo(`${device.room.name} - Mode: ${modeString} - Temperature scales:`)
+			platform.easyDebugInfo(`C:\n${JSON.stringify(modeCapabilities.temperatures.C, null, 4)}`)
+			platform.easyDebugInfo(`F:\n${JSON.stringify(modeCapabilities.temperatures.F, null, 4)}`)
 
-			if ('C' in modeCapabilities.temperatures || 'F' in modeCapabilities.temperatures) {
-				capabilities[mode].temperatures = {}
+			if (modeCapabilities.temperatures.C || modeCapabilities.temperatures.F) {
+				mode.temperatures = {}
 			}
 
 			// set min & max temperatures
 			if (modeCapabilities.temperatures?.C) {
-				capabilities[mode].temperatures.C = {
+				mode.temperatures.C = {
 					min: Math.min(...modeCapabilities.temperatures.C.values),
 					max: Math.max(...modeCapabilities.temperatures.C.values)
 				}
 			}
 
-			// TODO: check if we actaully need F, does Sensibo always return C if it has F?
+			// TODO: check if we actually need F, does Sensibo always return C if it has F?
 			if (modeCapabilities.temperatures?.F) {
-				capabilities[mode].temperatures.F = {
+				mode.temperatures.F = {
 					min: Math.min(...modeCapabilities.temperatures.F.values),
 					max: Math.max(...modeCapabilities.temperatures.F.values)
 				}
 			}
 
 			// set fanSpeeds
 			if (modeCapabilities.fanLevels && modeCapabilities.fanLevels.length) {
-				capabilities[mode].fanSpeeds = modeCapabilities.fanLevels
+				mode.fanSpeeds = modeCapabilities.fanLevels
 
 				// set AUTO fanSpeed
-				if (capabilities[mode].fanSpeeds.includes('auto')) {
-					capabilities[mode].autoFanSpeed = true
+				if (mode.fanSpeeds.includes('auto')) {
+					mode.autoFanSpeed = true
 				} else {
-					capabilities[mode].autoFanSpeed = false
+					mode.autoFanSpeed = false
 				}
 			}
 
 			// set vertical swing
 			if (modeCapabilities.swing) {
 				if (modeCapabilities.swing.includes('both')) {
-					capabilities[mode].horizontalSwing = true
-					capabilities[mode].verticalSwing = true
-					capabilities[mode].threeDimensionalSwing = true
+					mode.horizontalSwing = true
+					mode.verticalSwing = true
+					mode.threeDimensionalSwing = true
 				} else {
 					if (modeCapabilities.swing.includes('rangeFull')) {
-						capabilities[mode].verticalSwing = true
+						mode.verticalSwing = true
+						mode.threeDimensionalSwing = false
 					}
 
 					if (modeCapabilities.swing.includes('horizontal')) {
-						capabilities[mode].horizontalSwing = true
+						mode.horizontalSwing = true
+						mode.threeDimensionalSwing = false
 					}
 				}
 			}
 
 			// set horizontal swing
-			if (!capabilities[mode].horizontalSwing && modeCapabilities.horizontalSwing && modeCapabilities.horizontalSwing.includes('rangeFull')) {
-				capabilities[mode].horizontalSwing = true
+			if (!capabilities[modeString].horizontalSwing &&
```

Comment:
Based on your other changes shouldn't this be:
```suggestion
			if (!mode.horizontalSwing &&
```